backend/app/invoice_handler_test.go

```diff
@@ -0,0 +1,239 @@
+		var result map[string]interface{}
+		err := json.Unmarshal(resp.Body.Bytes(), &result)
+		assert.NoError(t, err)
+		assert.Equal(t, "Invoices are retrieved successfully", result["message"])
+	})
+
+	t.Run("Test List user invoices with no token", func(t *testing.T) {
+		req, _ := http.NewRequest("GET", "/api/v1/user/invoice/", nil)
+		resp := httptest.NewRecorder()
+		router.ServeHTTP(resp, req)
+		assert.Equal(t, http.StatusUnauthorized, resp.Code)
+	})
+
+	t.Run("Test List user invoices with empty list", func(t *testing.T) {
+		app2, err := SetUp(t)
+		require.NoError(t, err)
+		router2 := app2.router
+		err = app2.handlers.db.RegisterUser(user)
+		require.NoError(t, err)
+		token := GetAuthToken(t, app2, user.ID, user.Email, user.Username, false)
+		req, _ := http.NewRequest("GET", "/api/v1/user/invoice/", nil)
+		req.Header.Set("Authorization", "Bearer "+token)
+		resp := httptest.NewRecorder()
+		router2.ServeHTTP(resp, req)
+		assert.Equal(t, http.StatusOK, resp.Code)
+		var result map[string]interface{}
+		err = json.Unmarshal(resp.Body.Bytes(), &result)
+		assert.NoError(t, err)
+		assert.NotNil(t, result["data"])
+		data, ok := result["data"].(map[string]interface{})
+		assert.True(t, ok)
+		invoicesRaw, ok := data["invoices"]
+		assert.True(t, ok)
+		invoicesBytes, err := json.Marshal(invoicesRaw)
+		assert.NoError(t, err)
+		var invoices []models.Invoice
+		err = json.Unmarshal(invoicesBytes, &invoices)
+		assert.NoError(t, err)
+		assert.Len(t, invoices, 0)
+	})
+}
+
+func TestDownloadInvoiceHandler(t *testing.T) {
+	app, err := SetUp(t)
+	require.NoError(t, err)
+	router := app.router
+
+	user1 := CreateTestUser(t, app, "user1@example.com", "User One", []byte("securepassword"), true, false, 0, time.Now())
+
+	invoice := &models.Invoice{
+		ID:        1,
+		UserID:    user1.ID,
+		Total:     100.0,
+		Tax:       10.0,
+		CreatedAt: time.Now(),
+	}
+	err = app.handlers.db.CreateInvoice(invoice)
+	require.NoError(t, err)
+
+	t.Run("Download an invoice successfully", func(t *testing.T) {
+		token := GetAuthToken(t, app, user1.ID, user1.Email, user1.Username, false)
+		req, _ := http.NewRequest("GET", fmt.Sprintf("/api/v1/user/invoice/%d", invoice.ID), nil)
+		req.Header.Set("Authorization", "Bearer "+token)
+		resp := httptest.NewRecorder()
+		router.ServeHTTP(resp, req)
+		assert.Equal(t, http.StatusOK, resp.Code)
+		assert.Equal(t, "application/pdf", resp.Header().Get("Content-Type"))
+		assert.True(t, len(resp.Body.Bytes()) > 0)
+	})
+
+	t.Run("Download invoice with no token", func(t *testing.T) {
+		req, _ := http.NewRequest("GET", fmt.Sprintf("/api/v1/user/invoice/%d", invoice.ID), nil)
+		resp := httptest.NewRecorder()
+		router.ServeHTTP(resp, req)
+		assert.Equal(t, http.StatusUnauthorized, resp.Code)
+	})
+
+	t.Run("Download non-existing invoice", func(t *testing.T) {
+		token := GetAuthToken(t, app, user1.ID, user1.Email, user1.Username, false)
+		req, _ := http.NewRequest("GET", "/api/v1/user/invoice/99999", nil)
```

Comment:
suggestion: for the id we can get maximum ID and add 1 to it (this way it will be always non existing)